File: codeforces/icpc_2021_2022_latam/k.cpp
/**
 * K. KIARA is a Recursive Acronym
 * https://codeforces.com/gym/103640/problem/K
 */

// Accepted

#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>
#include <unordered_set>

using namespace std;

int main() {
    int n;
    cin >> n;

    unordered_set<char> first_letters;
    vector<string> words(n);

    for (int i = 0; i < n; ++i) {
        string word;
        cin >> word;

        first_letters.insert(word[0]);
        words[i] = word; 
    }

    bool one_is_all_in = false;

    for (const string& word : words) {
        bool all_in = true;

        for (char c : word) {
            if (first_letters.count(c) == 0) {
                all_in = false;
                break;
            }
        }

        if (all_in) {
            one_is_all_in = true;
            break;
        }
    }

    cout << ((one_is_all_in) ?  'Y' : 'N') << endl;
}

File: codeforces/icpc_2021_2022_latam/k.py
# TLE

def main():
    n = int(input())

    first_letters = set()
    words = []

    for _ in range(n):
        word = input()

        first_letters.add(word[0])
        words.append(word)

    one_was_valid = False

    for word in words:
        if all(c in first_letters for c in word):
            one_was_valid = True
            break

    print('Y' if one_was_valid else 'N')

main()
